crates/oxc_transformer/src/typescript/rewrite_extensions.rs

```diff
@@ -0,0 +1,78 @@
+use std::marker::PhantomData;
+
+use oxc_ast::ast::{
+    ExportAllDeclaration, ExportNamedDeclaration, ImportDeclaration, StringLiteral,
+};
+use oxc_traverse::{Traverse, TraverseCtx};
+
+use super::options::RewriteExtensionsMode;
+
+pub struct TypeScriptRewriteExtensions<'a> {
+    _marker: PhantomData<&'a ()>,
+    mode: RewriteExtensionsMode,
+}
+
+impl<'a> TypeScriptRewriteExtensions<'a> {
+    pub fn new(mode: RewriteExtensionsMode) -> Self {
+        Self { mode, _marker: PhantomData }
+    }
+
+    #[allow(clippy::case_sensitive_file_extension_comparisons)]
+    pub fn rewrite_extensions(&self, source: &mut StringLiteral<'a>, ctx: &mut TraverseCtx<'a>) {
+        let value = source.value.as_str();
+        if value.contains('/') || value.contains('\\') {
+            source.value = if value.ends_with(".ts") {
```

Comment:
I think you can match on `value.get(value.len() - 3..value.len())` 🤣

Reply:
Ahh, Yes! But should be `value.get(..value.len() - 3)` ?